src/Raven.Server/ServerWide/Maintenance/ClusterObserver.cs

```diff
@@ -642,9 +642,17 @@ private CompareExchangeTombstonesCleanupState GetMaxCompareExchangeTombstonesEta
                     if (hasState == false)
                         return CompareExchangeTombstonesCleanupState.InvalidDatabaseObservationState;
 
-                    if (nodeReport.Report.TryGetValue(state.Name, out var report) == false)
-                        continue;
+                    var tryGetValue = nodeReport.Report.TryGetValue(state.Name, out var report);
+                    Debug.Assert(tryGetValue, $"Could not find state for node '{nodeTag}' for database '{state.Name}'.");
+                    if (tryGetValue == false)
+                    {
+                        return CompareExchangeTombstonesCleanupState.InvalidDatabaseObservationState;
+                    }
```

Comment:
I encountered this scenario when I tried to create a test.
The test attempted to mimic the tombstone cleaning process on its own.

After discussing with @grisha-kotler, I understood that this scenario shouldn't happen. Therefore, I added a `Debug.Assert` here to ensure it will crash when running in debug mode, and I added a check to prevent it from crashing in production.